### PR TITLE
Fix data race on cid_arrival_map_.

### PR DESCRIPTION
**Pull request by HristoStaykov**
This is a fix for a reported corruption in cid_arrival_map_
from an ASAN run.

### CODE REVIEW

File: client/client_pool/src/concord_client_pool.cpp [hunk 1]
```diff
@@ -506,18 +506,21 @@ void ConcordClientPool::processReplies(ClientPtr &client,
...
-    cid_arrival_map_.erase(reply.cid);
-    duration = std::chrono::duration_cast<std::chrono::milliseconds>(before_send - arrival_time).count();
-    average_cid_receive_dur_.add(duration);
-
-    auto after_send = client->getAndDeleteCidResponseTime(reply.cid);
-    duration = std::chrono::duration_cast<std::chrono::milliseconds>(finish - after_send).count();
-    average_cid_close_dur_.add(duration);
+  {
+    std::unique_lock<std::mutex> lock(clients_queue_lock_);
+    metricsComponent_.UpdateAggregator();
```
Comment (yuliasherman) [inline client/client_pool/src/concord_client_pool.cpp:511]: Why does this need to be under a lock?

Comment (HristoStaykov) [reply client/client_pool/src/concord_client_pool.cpp]: I agree, I will move it out of the lock scope. Metrics have their own sync mechanism.

**Review by teoparvanov — APPROVED** [review]
LGTM, thanks Hristo!

**Review by yuliasherman — APPROVED** [review]
LGTM

**Pull request merged**